Dockerfile

```diff
@@ -6,12 +6,8 @@ RUN apk add --update git
 
 WORKDIR /plugin
 
-# warm up go mod cache
-COPY go.mod go.sum ./
-RUN go mod download
-
 COPY . /plugin
-RUN go build -v
+RUN go build -v -mod vendor
```

Comment:
Tbh, I am a bit new to Go. I am mostly copying from what I know from moby/swarmkit and moby/moby where I also added some PRs.

What is the reason for not vendoring vs vendoring?

Reply:
I try to avoid vendoring unless necessary: it kinda breaks some parts of the go trust-model. When using modules from the internet, you can kinda trust that everybody gets the same code for the dependencies (via mod proxy + sumdb, which can be audited). When using vendoring you - as an application consumer - can't be sure that the code inside `/vendor` is actually the same as the module name would indicate. You have to check.

Some people use vendoring as a way to speed up CI pipelines (IMHO not worth the trade-off) or to keep non-upstreamed changes local to the project (which is less work than upstreaming every change, but IMHO a bit short-sighted).